draft-ietf-privacypass-batched-tokens.md

```diff
@@ -317,15 +326,121 @@ struct {
 ~~~
 
 If the FinalizeBatch function fails, the Client aborts the protocol. Token
-verification works exactly as specified in {{ISSUANCE}}.
+verification works exactly as specified in {{RFC9578}}.
+
+# Arbitrary Batched Token Issuance
+
+This section describes an issuance protocol mechanism for issuing multiple tokens in one round trip between Client and Issuer. An arbitrary batched token request can contain token requests for any token type.
+
+## Client-to-Issuer Request {#arbitrary-client-to-issuer-request}
+
+The Client first creates all TokenRequest it wants to batch. To do so, the client follows
+protocol describing issuance, such as {{RFC9578, Section 5.1}} or {{RFC9578, Section 6.1}}.
+
+The Client then creates a BatchedTokenRequest structured as follows:
+
+~~~tls
+struct {
+  uint16_t token_type;
+  opaque token_request<0..2^16-1>;
+} TokenRequest
```

Comment:
```suggestion
struct {
   uint16_t token_type;
   select (token_type) {
      case (0x0001): /* Type VOPRF(P-384, SHA-384), RFC 9578 */
         uint8_t truncated_token_key_id;
         uint8_t blinded_msg[Ne];
      case (0x0002): /* Type Blind RSA (2048-bit), RFC 9578 */
         uint8_t truncated_token_key_id;
         uint8_t blinded_msg[Nk];
   }
} TokenRequest;
```
This isn't quite right. I think we just need to say that `TokenRequest` is simply prefixed by a uint16 value indicating the type, and that the rest of the structure follows based on that type.

Reply:
I've added this as an example. I have the impression the `opaque` structure better capture the general case, and the applicability to current and future token types

Reply:
`opaque token_request<0..2^16-1>;` means that the field `token_request` is length-prefix-encoded with two bytes, which is not how TokenRequest messages are structured. So it wasn't a matter of clarity, it just didn't match what's in the actual spec.